lib/sidekiq/rails.rb

```diff
@@ -3,6 +3,12 @@
 require "sidekiq/job"
 require "rails"
 
+begin
+  require "active_job"
+  require "active_job/queue_adapters/sidekiq_adapter"
```

Comment:
This will result in method redefinition warnings if running with a rails version that still ships the adapter, right? Would it make sense to add `gem "active_job", ">= 8.0.0"` to prevent this?

Reply:
Rails implicitly `autoload`s this class (https://github.com/rails/rails/blob/2f55eba02511ba3f79e4771305f14c85b1f7caf2/activejob/lib/active_job/queue_adapters.rb#L123) while we explicitly `require` it, so the one from rails will not be loaded and the redefinition should not happen?

Reply:
I didn't realize it works that way. I tested it out with your branch and it indeed seems fine.